File: server/models/business.js
const mongoose = require("mongoose");

// define schema of data
const businessSchema = new mongoose.Schema({
  lat: Number,
  lon: Number,
  address: String,
  name: String,
  description: String,
  imageUrl: String,
});

// exports schema to a model
const Business = mongoose.model("Business", businessSchema);

module.exports = Business;

File: lib/models/Business.dart
class Business {
  double lat;
  double lon;
  String address;
  String name;
  String description;
  String imageURL;
  String iconURL;

  Business({
    this.lat,
    this.lon,
    this.address,
    this.name,
    this.description,
    this.imageURL,
    this.iconURL,
  });

  factory Business.fromJSON(Map<String, dynamic> json) {
    final lat = json['lat'].toDouble();
    final lon = json['lon'].toDouble();
    final address = json['address'];
    final name = json['name'];
    final description = json['description'];
    final imageURL = json['imageURL'];

    return Business(
        lat: lat,
        lon: lon,
        address: address,
        name: name,
        description: description,
        imageURL: imageURL);
  }
}
